h2/src/main/org/h2/mvstore/MVStore.java

```diff
@@ -292,12 +292,21 @@ to a map (possibly the metadata map) -
         Object o = config.get("compress");
         this.compressionLevel = o == null ? 0 : (Integer) o;
         String fileName = (String) config.get("fileName");
+        fileStore = (FileStore) config.get("fileStore");
+        fileStoreIsProvided = fileStore != null;
+        if(fileStore == null && fileName != null) {
+            fileStore = new FileStore();
+        }
         o = config.get("pageSplitSize");
-        if (o == null) {
-            pageSplitSize = fileName == null ? 4 * 1024 : 16 * 1024;
+        int pgSplitSize;
+        if (o != null) {
+            pgSplitSize = (Integer) o;
+        } else if(fileStore != null) {
+            pgSplitSize = 16 * 1024;
         } else {
-            pageSplitSize = (Integer) o;
+            pgSplitSize = 48; // number of keys per page in that case
```

Comment:
eh, I merged it and then I noticed that you changed the value from 4 * 1024 to 48 - was that deliberate?